### PR TITLE
[SCSS/Bootstrap to Tailwind] .form-control and related stylings

### PR DESCRIPTION
**Pull request by mmmavis**
### Description

- Created a new Tailwind plugin `form-control.js`
- `form-control.js` is based on stylings on [Bootstrap v4.6's _form.scss](https://github.com/twbs/bootstrap/blob/v4.6.0/scss/_forms.scss)(only the relevant bits) and our custom SCSS overrides.
- Replace some classes with the newly created Tailwind ones. There are still some `.form-control` in the codebase that need to be updated to `.tw-form-control`. I've created a follow-up ticket for that so this PR doesn't grow too big #10253 

### To QA
1. Go to any campaign page with a petition form OR if you have staging db locally, go to `/en/campaigns/unknown-influence/`. Test the three forms on the page. There should not be any visual changes compared to what's on prod.
    - "newsletter" on nav
    - petition form (note that border radius and border colour have been updated. Changes are okayed by the design team)
    - footer form
2. Go to `/en/blog/` and test the newsletter form in the page (near footer)

### Related PRs/issues: 
  - #10252
  - #8704
  - #8335

### CODE REVIEW

**Review by fessehaye — COMMENTED** [review]
Does `form-control-feedback` have anything to do with the class `form-control` I do not see them using the same CSS inheritance. I do not even think this has to be apart of the plugin since it is used very little. I think we can replace some of those instances with just some utility classes and call it a day right?

**Review by fessehaye — CHANGES_REQUESTED** [review]
tw-has-danger is do not show up on IntelliSense so it is probably not being complied

File: tailwind-plugins/form-control.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        },
+        "select.form-control-lg": {
+          ...selectFormLgControl(theme),
+        },
+        "input[type='date'].form-control, input[type='time'].form-control, input[type='datetime-local'].form-control, input[type='month'].form-control":
+          {
+            appearance: "none",
+          },
+        ".has-danger": {
+          ...hasDanger(theme),
```
Comment (fessehaye) [inline tailwind-plugins/form-control.js:176]: You can address the `.has-danger` class in two different ways. You can make a danger variant and address every HTML instance of `form-control` and utility classes. Or you can address has-danger the same way we kinda address adding dark class for form-control in L117. I do not think the has-danger function work in this case.

**Comment by fessehaye** [comment]
When we use the `addComponents` API, we want to be able to use one CSS class and hover over that class and get all of the information we need about what is being included. In your example, we do that for `.has-danger` it doesn't really make sense to make a component for it.

When we are converting SASS files we have to keep in mind that `@extend` feature is used quite a bit and if we want to transition out of SASS or move it to a tailwind plugin we usually have to flatten the SASS. There will be some situations where it feels too awkward to remove certain SCSS lines but if that is the case keeping some remaining SASS is totally fine as long as we try to remove SCSS rules. That is the most essential part about cleaning the SASS files since we can always further clean up the remaining CSS at a later time after the conversion to tailwind is finished.

File: tailwind-plugins/form-control.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+  paddingX: "12px",
+  paddingRight: "48px", // paddingX * 2 + backgroundSize
+};
+
+/**
+ * Stylings for .form-control (converted from Bootstrap 4.6 with our custom SCSS)
+ * @param {*} theme Tailwind theme config
+ * @returns {object} Stylings
+ */
+function formControl(theme) {
```
Comment (fessehaye) [inline tailwind-plugins/form-control.js:17]: Thank you for using the theme variables.

**Comment by fessehaye** [comment]
This is almost there @mmmavis. Just a few corrections we need to address. If you need any more help let me know. Thanks for doing this

File: tailwind-plugins/form-control.js (hunk 1 above)
Comment (mmmavis) [reply tailwind-plugins/form-control.js:176]: Thanks for the suggestion. Can you elaborate on `I do not think the has-danger function work in this case.`? I checked again and verified the `.tw-has-danger .form-control` and `.tw-has-danger .form-control-feedback` DOM elements are having the stylings I defined here.

Comment (fessehaye) [reply tailwind-plugins/form-control.js:176]: image
When you hover over `.tw-form-control` it reveals zero information about the has-danger classes. 

Hovering over `tw-has-danger` does nothing as well and should not be a component.

**Comment by fessehaye** [comment]
> > Does `form-control-feedback` have anything to do with the class `form-control` I do not see them using the same CSS inheritance. I do not even think this has to be apart of the plugin since it is used very little. I think we can replace some of those instances with just some utility classes and call it a day right?
> 
> Thanks for pointing out that `.form-control-feedback` doesn't rely on `.form-control` stylings. I can remove it from `form-control.js` and use utility classes inline instead. However, I do have a question - with this approach, how do we ensure future "form feedback" elements are using the same stylings? Does it require a dev to dig into the codebase, find a form that has "feedback" element, and grab the same set of utility classes to use again?

Yeah, good question! If we can remove a CSS class name it is always the better move. A lot of them do not have consistent naming conventions or at least ones we have not been consistent about. Along the way you will probably find css-class names that could just be built with tailwind classes. The exception is special cases like CMS content or tricky CSS rules/conditions.

Yeah, the goto is just digging the codebase. Sometimes you will find cases were refactoring the HTML could save you time,but it might be out of scope for these smaller tickets.

File: tailwind-plugins/form-control.js (hunk 1 above)
Comment (mmmavis) [reply tailwind-plugins/form-control.js:176]: @fessehaye I think the above problem is actually from Tailwind CSS IntelliSense. I tried to comment out `...formControlPlugins,` in `tailwind.config.js` and uncomment it again, everything then got generated as expected.

image

**Comment by mmmavis** [comment]
@fessehaye PR updated! See individual commits for details.

File: tailwind-plugins/form-control.js (hunk 1 above)
Comment (fessehaye) [reply tailwind-plugins/form-control.js]: interesting, I guess since we are moving away from this implementation of has-danger, it doesn't matter atm

**Review by fessehaye — APPROVED** [review]

**Pull request merged**